### PR TITLE
feat: handle backend error adding allowances

### PR DESCRIPTION
**Pull request by Zacharis278**
[COSMO-418](https://2u-internal.atlassian.net/browse/COSMO-418)

Adds error display for when a request to add allowances is rejected by the backend. This would typically happen because a username/email does not exist.

Screenshot 2024-08-08 at 12 22 52 PM

### CODE REVIEW

File: src/data/redux/hooks/requests.js [hunk 1]
```diff
@@ -29,15 +29,20 @@ export const useMakeNetworkRequest = () => {
       if (onSuccess) { onSuccess(response); }
       dispatch(actions.completeRequest({ requestKey, response }));
     }).catch((error) => {
-      if (onFailure) { onFailure(error); }
-      dispatch(actions.failRequest({ requestKey, error }));
+      // Parse backend error from axios object
```
Comment (Zacharis278) [inline src/data/redux/hooks/requests.js:32]: As is this is always going to be an axios object but in the event we do some custom error parsing before we get here the error will pass through.

File: src/pages/ExamsPage/hooks.js [hunk 2]
```diff
@@ -3,7 +3,7 @@ import { useSelector, useDispatch } from 'react-redux';
 
 import { RequestKeys } from 'data/constants';
 
-import * as reduxHooks from 'data/redux/hooks/requests';
+import * as reduxHooks from 'data/redux/hooks';
```
Comment (Zacharis278) [inline src/pages/ExamsPage/hooks.js:6]: The contents of `/requests` in already exported from an index file for that module. Got tripped up in my tests by that inconsistency so this is just a bit of clean up.

**Review by rijuma — APPROVED** [review]
LGTM

**Review by schenedx — APPROVED** [review]
👍

**Pull request merged**